### ISSUE TITLE
Rule: no-unused-placeholders

### ISSUE THREAD

**Issue by not-an-aardvark** [not-an-aardvark/eslint-plugin-eslint-plugin#7]
Reports when a `context.report` call contains a `data` property that does not have a corresponding placeholder in the report message.

Example of invalid code:

```js
context.report({
  node,
  message: "Do not use {{foo}}.",
  data: {
    foo: 1,
    bar: 2
  }
});
```

**Comment by Aladdin-ADD** [comment]
is this proper to add to rule no-missing-placeholders? they are so similar!

**Comment by not-an-aardvark** [comment]
I don't have a strong opinion either way. I think it would be fine to have them as two separate rules because they are checking two separate things (one of them checks missing placeholders, and one of them checks missing `data`) but I'm also fine with combining them.
